src/doc/src/reference/specifying-dependencies.md

```diff
@@ -2,88 +2,113 @@
+version `0.1.13` if it is the latest `0.1.z` release, but would not update us
+to `0.2.0`.
 
-```notrust
-1.2.3  :=  >=1.2.3, <2.0.0
-1.2    :=  >=1.2.0, <2.0.0
-1      :=  >=1.0.0, <2.0.0
-0.2.3  :=  >=0.2.3, <0.3.0
-0.2    :=  >=0.2.0, <0.3.0
-0.0.3  :=  >=0.0.3, <0.0.4
-0.0    :=  >=0.0.0, <0.1.0
-0      :=  >=0.0.0, <1.0.0
-```
+There are five types of version requirements:
 
-This compatibility convention is different from SemVer in the way it treats
-versions before 1.0.0. While SemVer says there is no compatibility before
-1.0.0, Cargo considers `0.x.y` to be compatible with `0.x.z`, where `y ≥ z`
-and `x > 0`.
+| Type                                   | Operators         | Examples                               |
+|----------------------------------------|-------------------|----------------------------------------|
+| [Caret](#caret-requirements)           | None or `^`       | `"1.2.3"` and `"^1.2.3"`               |
+| [Tilde](#tilde-requirements)           | `~`               | `"~1.2.3"`                             |
+| [Wildcard](#wildcard-requirements)     | `*`               | `"1.2.*"`                              |
+| [Comparison](#comparison-requirements) | `>`, `<` and `>=` | `">1.2.3"`, `"<1.2.3"` and `">=1.2.3"` |
+| [Strict](#strict-requirements)         | `=`               | `"= 1.2.3"`                            |
 
-It is possible to further tweak the logic for selecting compatible versions
-using special operators, though it shouldn't be necessary most of the time.
+[SemVer]: https://doc.rust-lang.org/cargo/reference/resolver.html#semver-compatibility
 
 ### Caret requirements
 
-**Caret requirements** are an alternative syntax for the default strategy,
-`^1.2.3` is exactly equivalent to `1.2.3`.
+**Caret requirements** allow SemVer compatible updates.
+
+Updates are only allowed if the new version number does not modify the left-most
+non-zero digit.
+
+This compatibility convention is different from [SemVer] in the way it treats
+versions before `"1.0.0"`. While SemVer says there is no compatibility before
+1.0.0, Cargo considers `"0.x.y"` to be compatible with `"0.x.z"`, where `y ≥ z`
+and `x` is not zero.
+
+#### Examples:
+
+If we specify the version string as `"1.4.8"`, compatible versions are:
+- `"1.4.x"` with `x ≥ 8`.
+- `"1.x.y"` with `x > 2`, `y` can be anything.
+
+This table shows all possible ways to specify a caret requirements, and the
+ranges that are considered compatible for each:
+
+```notrust
+^1      :=  >=1.0.0, <2.0.0
+^1.2    :=  >=1.2.0, <2.0.0
+^1.2.3  :=  >=1.2.3, <2.0.0
+^0      :=  >=0.0.0, <1.0.0
+^0.0    :=  >=0.0.0, <0.1.0
+^0.0.3  :=  >=0.0.3, <0.0.4 (exact)
+^0.2    :=  >=0.2.0, <0.3.0
+^0.2.3  :=  >=0.2.3, <0.3.0
+```
+
+> **Note**: This is the only type of version requirement that has different
+> rules for versions that start with `0`, and the version `"0.x.y"` is not
+> compatible with any other versions.
```

Comment:
I'm not sure I understand this last clause about 0.x.y not being compatible with anything else, isn't 0.x.(y+1) compatible with that?

Reply:
Yes, if `x > 0`.

Before the review this was stating that `0.0.x` was not compatible with other versions, @ehuss said that `0.0.x` was not a common version, so it wouldn't be as useful in this explanation, so I changed to `0.x.y` without noticing that it is now incorrect.

I'm not sure how to proceed here, is this information even useful? I do often get confused about this, maybe I should just leave the definition.

Or remove completely the "and" part?

> , and the version `"0.x.y"` is not compatible with any other versions.

Reply:
What is this note trying to convey? It's correct that `0.0.x` is not that common but it's incorrect to say that `0.x.y` is not compatible with any other versions.